src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -117,86 +118,116 @@ public bool TryRemoveContext(IAzureContext context)
+                            else
+                            {   // if account ID is different from the first tenant account id we need to ignore current tenant
+                                WriteWarningMessage(string.Format(
+                                    ProfileMessages.AccountIdMismatch,
+                                    account.Id,
+                                    tenant,
+                                    accountId));
+                                account.Id = accountId;
+                                token = null;
+                            }
                         }
-                        else if (accountId.Equals(account.Id, StringComparison.OrdinalIgnoreCase))
+                        catch
                         {
-                            account.SetOrAppendProperty(AzureAccount.Property.Tenants, tenant);
+                            WriteWarningMessage(string.Format(ProfileMessages.UnableToAqcuireToken, tenant));
                         }
-                        else
-                        {   // if account ID is different from the first tenant account id we need to ignore current tenant
-                            WriteWarningMessage(string.Format(
-                                ProfileMessages.AccountIdMismatch,
-                                account.Id,
-                                tenant,
-                                accountId));
-                            account.Id = accountId;
-                            token = null;
-                        }
-                    }
-                    catch
-                    {
-                        WriteWarningMessage(string.Format(ProfileMessages.UnableToAqcuireToken, tenant));
-                    }
 
-                    if (token != null &&
-                        newTenant == null &&
-                        TryGetTenantSubscription(token, account, environment, tenant, subscriptionId, subscriptionName, out tempSubscription, out tempTenant))
-                    {
-                        // If no subscription found for the given token/tenant 
-                        // discard tempTenant value unless current token/tenant is the last one.
-                        if (tempSubscription != null || i == (tenants.Count() - 1))
+                        if (token != null &&
+                            newTenant == null &&
+                            TryGetTenantSubscription(token, account, environment, tenant, subscriptionId, subscriptionName, out tempSubscription, out tempTenant))
                         {
-                            newTenant = tempTenant;
-                            newSubscription = tempSubscription;
+                            // If no subscription found for the given token/tenant 
+                            // discard tempTenant value unless current token/tenant is the last one.
+                            if (tempSubscription != null || tenant.Equals(tenants[tenants.Count]))
```

Comment:
won't this be one past the end of the tenant list?

Reply:
Yeah, you want `tenant == tenants[tenants.Count - 1]`.

Why do we lean towards `.Equals`?

Reply:
Thanks for the catch guys. @twitchax I lean towards .Equals because that's how you are supposed to compare strings in Java.  Is it different in C#?

Reply:
Ahhh, so, it is a bit different in C#.  `==` is normally preferred, but you might want to make it clear that you _care_ about reference equality.  In that case, it may be best to make it explicit by doing `object.ReferenceEquals(tenant, tenants[tenants.Count - 1]`.

For strings, `==` is a value comparison, and `.Equals` is a reference compare.  Either will work, but you're right, the reference compare is better.